src/handlers/Handler.ts

```diff
@@ -2,7 +2,22 @@ import HandlerConfig from "./HandlerConfig";
 import { GroupMeInfo } from "../services/GroupMeService";
 import { Either } from "fp-ts/lib/Either";
 
-export default interface Handler {
-  readonly config: HandlerConfig;
-  handle(groupMeInfo: GroupMeInfo): Promise<Either<Error, unknown>>;
+export default abstract class Handler<R = false> {
+  protected subHandlers: Handler[] = [];
+
+  readonly config: R extends false ? HandlerConfig : never;
```

Comment:
What is this exactly?

Reply:
you talking about the type on config?

Reply:
R defaults to false and `config` extends `HandlerConfig` if only `true`?

Reply:
Comment needed, but essentially if you are the root handler you don't need a config is what I was going for.

If the root handler should have a config then I can amend it.